akka-persistence-query/src/test/scala/akka/persistence/query/typed/internal/EventsBySliceFirehoseSpec.scala

```diff
@@ -223,12 +223,15 @@ class EventsBySliceFirehoseSpec
       // FIXME is there a way to know that it has been added to the hub?
       Thread.sleep(1000)
 
-      firehosePublisher.sendNext(allEnvelopes(0))
       catchupPublisher(0).sendNext(allEnvelopes(0))
       catchupPublisher(1).sendNext(allEnvelopes(0))
       outProbe(0).expectNext(allEnvelopes(0))
       outProbe(1).expectNext(allEnvelopes(0))
 
+      firehosePublisher.sendNext(allEnvelopes(0))
+      // this "sleep" is needed so that the firehose envelope is received first
+      outProbe.expectNoMessage()
```

Comment:
not great that it's timing sensitive, but don't know an easy way to solve that

Reply:
I don't think we can do better than that.